modules/minimint-mint/src/lib.rs

```diff
@@ -482,8 +486,8 @@ impl Mint {
         }
 
         // Determine the reference response to check against
-        let our_contribution = match &partial_sigs.iter().find(|(peer, _)| *peer == self.key_id) {
-            Some((_, psigs)) => psigs,
+        let our_contribution = match our_contribution {
+            Some(psig) => psig,
```

Comment:
I think now that we actually check against the right piece of data we should look into marking peers that submit shares not belonging to any issuance as faulty.

This would require a bit more work though. Currently there could be a share that's submitted too late and we already deleted our proposal. I think what we should do is fetch `or_contribution` from either the proposals "table" or the agreed "table". That would also allow us to delete our share as soon as it was in a consensus output. (see below)

Reply:
Sure, I don't think that's difficult to add, although sending random contributions to peers doesn't seem like an attack vector that could reduce max_evil (compared to not contributing to valid issuances)?

Reply:
Good point, it only marginally slows down consensus, so should warn but not ban in case it's because we built a bug into the system.